src/components/logic/Quiz.js

```diff
@@ -0,0 +1,144 @@
+import React, { useState, useRef, useContext } from 'react'
+import Tally from '../actions/Tally'
+import { Rounds } from '../Context'
+import QuizQuestion from '../views/QuizQuestion'
+
+
+export default function Quiz ({ data, round }) {
+  const [rounds, updateRounds] = useContext(Rounds)
+  const [currentQ, nextQ] = useState(data[0].questions[0])
+  const [colors, setColors] = useState([])
+  const thisInput = useRef(null)
+  const currentChord = useRef(data[0])
+  const roundData = useRef([])
+  const chord = useRef({
+    chord: data[0].notes, //this is not a unique id. we should decide on one when we work on the data structure.
+    questions: []
+  })
+  const subQ = useRef({
+    type: data[0].questions[0].type,
+    answers: []
+  })
+  const answer = useRef({
+    answer: data[0].questions[0].answers[0],
+    tries: [],
+    startTime: Date.now(),
+    endTime: '',
+    elapsedTime: '',
+  })
+
+  function handleClick(input) {
+    if (thisInput.current === false) {
+      return
+    }
+    answer.current.tries.push({'input': input, type: 'click'})
+    thisInput.current = input
+    checkInput(input)
+  }
+  function onKeyPressed(e) {
+    if (thisInput.current === false) {
+      return
+    }
+    const key = e.key
+    const input = (() => {
+      for (let choice of currentQ.choices) {
+        if (key === choice.key) {
+          return choice.choice
+        }
+      }
+      return null
+    })()
+    if (input !== null) {
+      answer.current.tries.push({'input': input, type: 'keypress'})
+      thisInput.current = input
+    }
+    checkInput(input)
+  }
+  function checkInput(input) {
+    switch (input) {
+      case null:
+        return
+      case (currentQ.answers[subQ.current.answers.length]):
+        answer.current.endTime = Date.now()
+        answer.current.elapsedTime = (answer.current.endTime-answer.current.startTime)/1000
+        subQ.current.answers.push(answer.current)
+        answer.current = {
+          answer: data[roundData.current.length].questions[chord.current.questions.length].answers[subQ.current.answers.length],
+          tries: [],
+          startTime: Date.now(),
+          endTime: '',
+          elapsedTime: '',
+        }
+        return setNextView(input)
+      default:
+        return setColors([...colors, {'input': input, color: 'red'}])
```

Comment:
@jsbean here's where the new consolidated `colors` object/logic is working if you didn't see it before